imap_processing/glows/l0/decom_glows.py

```diff
@@ -64,11 +65,28 @@ def decom_packets(
                 apid = packet.header["PKT_APID"].derived_value
                 # Do something with the packet data
                 if apid == GlowsParams.HIST_APID.value:
-                    hist_l0 = HistogramL0(packet, version, filename)
+                    values = [
+                        item.derived_value
+                        if item.derived_value is not None
+                        else item.raw_value
+                        for item in packet.data.values()
+                    ]
+                    hist_l0 = HistogramL0(
```

Comment:
I think I am starting to understand the dataclass after working with CDF attrs. So here, you are creating instance of HistogramL0 to store current packet's data and then add that to the list. So the list of unpacked data is list of instances of HistogramL0. same for other apid. Just thinking out loud here.

Reply:
Exactly right. That way there is a check here for missing or misaligned attributes, so if for whatever reason the packet is misformed it will break right away. Then, I can pass around the dataclass instance without worrying that, for example, it's a dictionary that can have items accidentally deleted out of it. I can also easily access attributes by just typing `l0.HISTOGRAM_DATA`, so I don't need to remember or worry about ordering in a dictionary. It also tracks the expected data type, and of course, I can put any level-specific methods right into each instance, although in this case it's not super useful, and of course for L0 I don't really have many.